tests/integration/docker_utils/test_docker.py

```diff
@@ -1444,6 +1445,49 @@ def test_run_with_ulimit(self, docker_client: ContainerClient):
         )
         assert stdout.decode(config.DEFAULT_ENCODING).strip() == "1024"
 
+    def test_run_with_additional_arguments_env_files(
+        self, docker_client: ContainerClient, tmp_path
+    ):
+        env_variable = "TEST1=VAL1"
+        env_file = tmp_path / "env1"
+        env_vars = textwrap.dedent("""
+            # Some comment
+            TEST1=OVERRIDDEN
+            TEST2=VAL2
+            TEST3=${TEST2}
+            TEST4=VAL # end comment
+            """)
+        env_file.write_text(env_vars)
+
+        stdout, _ = docker_client.run_container(
+            "alpine",
+            remove=True,
+            command=["env"],
+            additional_flags=f"-e {env_variable} --env-file {env_file}",
+        )
+        env_output = stdout.decode(config.DEFAULT_ENCODING)
+        # behavior differs here from more advanced env file parsers
+        assert env_variable in env_output
+        assert "TEST1=VAL1" in env_output
+        assert "TEST2=VAL2" in env_output
+        assert "TEST3=${TEST2}" in env_output
+        assert "TEST4=VAL # end comment" in env_output
```

Comment:
I assume those validations specifically would be working properly with other parsers right?

Reply:
With a proper env file parser, the end of line comment would be stripped, and the variable would be substituted. However, docker does not do it, so neither should we I guess 🤷

Reply:
Love those parity situations where we need to lower our feature set so we aren't too powerful :grin: